CONTRIBUTING.md

```diff
@@ -55,13 +58,34 @@ often much easier to reason about than describing a problem and solution.
 
 That said, we have a number of bars for any contribution to clear:
 
-- It must pass the entire test suite and linting (including gofmt)
+- It must pass the entire test suite and linting (including gofmt).
 - If it introduces a new feature or changes an existing feature, that feature
-  must be documented (how else will people find out about your awesome work?)
+  must be documented (how else will people find out about your awesome work?).
 - It can't change existing syntax except in extremely well-reasoned cases. If
   you're changing the syntax of existing features, definitely open an issue
   first so we can discuss.
 
+#### Test Coverage
+
+##### Our Testing Goals
+
+We want to be sure of a few things, and testing will help us accomplish this:
+
+- Help ensure a new feature or changes to an existing feature are properly
+implemented.
+- Avoid introducing bugs.
+- Avoid regressions when a new feature is introduced.
+
+##### Tests
+
+Adequate testing can vary based on the change being introduced. If you have
+questions, don't hesitate to ask. In general, this is how we'd like tests to be
+addressed:
+
+- Include tests for the way your code interacts with the core engine.
+- Test functionality of the code you have introduced, say in a new module.
+- Mock interaction with the system.
```

Comment:
I think if we're going to tell people "you shouldn't mock, except when you should, but usually not" it's probably better to just not mention it at all and leave it at the fact that we should have tests for things.